torchgeo/datasets/oscd.py

```diff
@@ -432,17 +432,25 @@ def setup(self, stage: Optional[str] = None) -> None:
         def n_random_crop(sample: Dict[str, Any]) -> Dict[str, Any]:
             images, masks = [], []
             for i in range(self.num_patches_per_tile):
-                mask = repeat(sample["mask"], "h w -> t h w", t=2)
+                mask = repeat(sample["mask"], "h w -> t h w", t=2).float()
                 image, mask = self.rcrop(sample["image"], mask)
                 mask = mask.squeeze()[0]
                 images.append(image.squeeze())
-                masks.append(mask)
+                masks.append(mask.long())
             sample["image"] = torch.stack(images)
             sample["mask"] = torch.stack(masks)
             return sample
 
+        def pad_to(sample: Dict[str, Any]) -> Dict[str, Any]:
+            padto = K.PadTo((1280, 1280))
```

Comment:
This is instantiating a K.PadTo object on each call. We should make this a class attribute.